src/Chapter3.hs

```diff
@@ -970,7 +1049,21 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data Weekday = Mon | Tue | Wed | Thu | Fri | Sat | Sun deriving (Show, Eq, Enum, Bounded)
+
+isWeekend :: Weekday -> Bool
+isWeekend weekday = case weekday of
+  Sat -> True
+  Sun -> True
+  _ -> False
+
+nextDay :: Weekday -> Weekday
+nextDay weekday
+  | weekday == maxBound = minBound -- roll over to start of week
+  | otherwise = succ weekday
 
+daysToParty :: Weekday -> Int
+daysToParty weekday = abs (fromEnum weekday - fromEnum Fri)
```

Comment:
Unfortunately, time doesn't work this way, and we can't travel back in time :disappointed: 
For example, if today is Saturday, there are 6 days to the next Friday, not 1 (as in your current solution).